cluster-autoscaler/provisioningrequest/checkcapacity/provisioningclass.go

```diff
@@ -95,130 +95,123 @@ func (o *checkCapacityProvClass) Provision(
 	nodeInfos map[string]*framework.NodeInfo,
 ) (*status.ScaleUpStatus, errors.AutoscalerError) {
 	combinedStatus := NewCombinedStatusSet()
-	provisioningRequestsProcessed := make(map[string]bool)
 	startTime := time.Now()
 
 	o.context.ClusterSnapshot.Fork()
 	defer o.context.ClusterSnapshot.Revert()
 
-	prPods := unschedulablePods
+	// Gather ProvisioningRequests.
+	prs, err := o.getProvisioningRequestsAndPods(unschedulablePods)
+	if err != nil {
+		return status.UpdateScaleUpError(&status.ScaleUpStatus{}, errors.NewAutoscalerError(errors.InternalError, "%s", err.Error()))
+	} else if len(prs) == 0 {
+		return &status.ScaleUpStatus{Result: status.ScaleUpNotTried}, nil
+	}
 
-	for len(prPods) > 0 {
-		prs := provreqclient.ProvisioningRequestsForPods(o.client, prPods)
-		prs = provreqclient.FilterOutProvisioningClass(prs, v1.ProvisioningClassCheckCapacity)
-		if len(prs) == 0 {
-			break
-		}
+	// Add accepted condition to ProvisioningRequests.
+	for _, pr := range prs {
+		conditions.AddOrUpdateCondition(pr.PrWrapper, v1.Accepted, metav1.ConditionTrue, conditions.AcceptedReason, conditions.AcceptedMsg, metav1.Now())
```

Comment:
Previous marking as accepted mechanism sets `lastProvisioningRequestProcessTime` in injector:
https://github.com/kubernetes/autoscaler/blob/7c0146a6d86f5b12373fa64c62a1caf826c7147f/cluster-autoscaler/processors/provreq/injector.go#L76-L84
Shouldn't the logic be kept in some way?

Reply:
I added a (hacky) method to the injector, which we will call in Provision, if we found at least 1 PR to process.